src/DocumentFeedMigrator.cs

```diff
@@ -4,22 +4,27 @@
 
 namespace MigrationExecutorFunctionApp
 {
+    using Microsoft.Azure.Cosmos;
+    using Microsoft.Azure.Documents;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
     using System;
     using System.Collections.Concurrent;
     using System.Collections.Generic;
     using System.Diagnostics;
     using System.Linq;
+    using System.Text;
     using System.Threading;
     using System.Threading.Tasks;
-    using Microsoft.Azure.Cosmos;
-    using Microsoft.Azure.Documents;
-    using Microsoft.Azure.WebJobs;
-    using Microsoft.Extensions.Logging;
+
+
 
     public class DocumentFeedMigrator
     {
         private Container containerToStoreDocuments;
-
+        string targetPartitionKeyAttribute = Environment.GetEnvironmentVariable($"{"TargetPartitionKeyAttribute"}");
+        static string sourcePartitionKeyMapping = Environment.GetEnvironmentVariable($"{"SourcePartitionKeyMapping"}");    
+        Boolean isSyntheticKey = Environment.GetEnvironmentVariable($"{"SourcePartitionKeyMapping"}").IndexOf(",") == -1? false: true;
```

Comment:
With this change you are now extracting the PK even on a scenario where the destination PK is not synthetic, which doesn't add any value.